ENIGMAsystem/SHELL/Universal_System/roomsystem.cpp

```diff
@@ -250,15 +250,15 @@ namespace enigma
         dit->second.tiles.clear();
       }
     }
-    for (int i = 0; i < tilecount; i++) {
+    for (size_t i = 0; i < tiles.size(); i++) {
       tile t = tiles[i];
       drawing_depths[t.depth].tiles.push_back(t);
     }
     load_tiles();
     //Tiles end
 
-    object_basic* is[instancecount];
-    for (int i = 0; i < instancecount; i++) {
+    std::vector<object_basic*> is(instances.size());
```

Comment:
I dunno what dafux you want there. What is kept and what is deleted in ... If you want it changed give me a full diff

Reply:
besides the one you changedalso theres like 4 or 5 index based loops there other than the one you changed

Reply:
I was asking you to update this so that it's not wasting space and time iterating a bunch of nulls. It's probably going to be solid, but there's not a compelling reason to ditch the old double-length system entirely and just use `push_back` and new ranged iteration.

Reply:
your code doesn't even compile. If you want it replaced either replace it or give me correct code. The point of this change is not to tidy your old code

Reply:
Josh is making a good point here, but he did make a few typos in the code. Here's my corrected version, `obj` needs accessed like a reference, not a pointer, and the created instances should be pushed to created not instances.
```cpp
    std::vector<object_basic*> created;
    created.reserve(instances.size());
    for (const inst &obj : instances)
      if (!enigma::fetch_instance_by_id(obj.id))
        created.emplace_back(instance_create_id(obj.x,obj.y,obj.obj,obj.id));
```

Reply:
Ok: b7e0dd36101dbd4c20d45b1e8cf19da682f0a69d

Apparently I still have direct write access to his fork, so I just did it myself.